analyses/survival-analysis/survival-analysis_histology.Rmd

```diff
@@ -0,0 +1,257 @@
+                                      ggtheme = theme_bw())
+  print(ggpar(plot_coxph[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_coxph[[2]], font.legend=6))
+
+  dev.off()
+
+  print(ggpar(plot_coxph[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_coxph[[2]], font.legend=6))
+}
+
+```
+
+## Log Rank analysis 
+### Generate output for categorical files 
+```{r}
+ind_var_list <- c("hgg_group", "broad_histology_display")
+
+for(i in 1:length(ind_var_list)){
+  ind_var <- ind_var_list[i]
+
+  # define model
+  model <- paste0("survival::Surv(time = OS_years, event = OS_status_fixed) ~ ", ind_var)
+  
+  # run survival analysis
+  fit <- survival::survdiff(formula(model),
+                            data = meta_indep)
+  # Obtain p value for Chi-Squared stat
+  fit$p.value <- pchisq(fit$chisq, df = length(fit$n) - 1, lower = FALSE)
+  
+  print(fit)
+  # save the output
+  saveRDS(fit, file.path(results_dir, paste0("log_rank_survival_per_", ind_var, ".RDS")))
+  
+  # generate plots fit
+  fit_plot <- survfit(formula(model), data = meta_indep)
+  
+  # output the plot
+  pdf(file.path(plots_dir, paste0("logrank_survival_per_", ind_var, ".pdf")))
+  plot_logrank <- survminer::ggsurvplot(fit_plot,
+                                        data=meta_indep,
+                                        xlim = c(0, 14),
+                                        break.time.by = 1,
+                                        pval = TRUE, 
+                                        conf.int = TRUE,
+                                        risk.table = TRUE, # Add risk table
+                                        linetype = "strata", # Change line type by groups
+                                        surv.median.line = "hv", # Specify median survival
+                                        ggtheme = theme_bw())
+  print(ggpar(plot_logrank[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_logrank[[2]], font.legend=6))
+
+  dev.off()
+
+  print(ggpar(plot_logrank[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_logrank[[2]], font.legend=6))
+}
+
+```
+
+## Multivariate analysis 
+### Two comparisons made are `tp53_score + telomerase_score + hgg_group` and `tp53_score + telomerase_score + broad_histology_display`
+
+```{r}
+for(ind_var in c("tp53_score+telomerase_score+hgg_group", "tp53_score+telomerase_score+broad_histology_display")){
+  model <- paste0("survival::Surv(time = OS_years, event = OS_status_fixed) ~ ", ind_var)
+  
+  fit <- survival::coxph(
+        formula(model),
+        data = meta_indep
+      )
+  # generate output
+  table <- broom::tidy(fit) 
+  
+  # Save the table data in a TSV
+  readr::write_tsv(table, file.path(results_dir, paste0("cox_reg_results_per_", ind_var, ".tsv")))
+  
+  print(table)
+  
+  # output the plot
+  forest_coxph <- survminer::ggforest(fit, data = meta_indep)
+  print(forest_coxph)
+}
+
+```
+
+## Bivariate density plot of tp53 score and telomerase activity score
```

Comment:
Awesome -- I think this is really helpful to show why the multivariate model is necessary in addition to the univariate.